pkg/block/index.go

```diff
@@ -524,6 +548,66 @@ func IgnoreDuplicateOutsideChunk(_, _ int64, last, curr *chunks.Meta) (bool, err
 	return true, nil
 }
 
+// FixDuplicateSamplesOutsideChunk takes in two pointers to chunks.Meta structs,
+// last and curr, and returns a new chunks.Meta struct and an error. If last is
+// nil, the function returns curr and nil. Otherwise, it creates a new empty
+// chunk based on the encoding of curr.Chunk.
+//
+// Parameters:
+// - last: a pointer to the last chunk.Meta struct
+// - curr: a pointer to the current chunk.Meta struct
+//
+// Return:
+// - *chunks.Meta: a pointer to the new chunk.Meta struct
+// - error: an error if any occurred during the function execution.
+func FixDuplicateSamplesOutsideChunk(last, curr *chunks.Meta) (*chunks.Meta, error) {
+	if last == nil {
+		return curr, nil
+	}
+
+	c, err := chunkenc.NewEmptyChunk(curr.Chunk.Encoding())
+	if err != nil {
+		return curr, err
+	}
+	ncurr := &chunks.Meta{
+		MinTime: int64(0),
+		MaxTime: curr.MaxTime,
+		Chunk:   c,
+	}
+	if curr.MinTime == last.MaxTime {
+		var it chunkenc.Iterator
+		it = curr.Chunk.Iterator(it)
+		app, _ := ncurr.Chunk.Appender()
+		typ := it.Next()
+		for ; typ != chunkenc.ValNone; typ = it.Next() {
```

Comment:
Can we simplify it to

```
var (
  t int64
  v float64
)
for it.Next() == chunkenc.ValFlot {
  t, v = it.At()
  if t <= last.MaxTime {
    continue
  }
  app.Append(t, v)
}
```

Reply:
I checked the promehteus bug source code and found nothing that would cause a histogram. I was also worried about this problem, so I reserved space in case such problems occurred in the future.